### ISSUE TITLE
Wrong salt-minion's main process name when python-setproctitle installed

### ISSUE THREAD

**Issue by avlasov-mos-de** [saltstack/salt#43316]
### Description of Issue/Question
salt-minion tries to import module setproctitle. If this done successfully it executes  setproctitile  with space instead of "\0" used as param separator. 

```python
try:
    import setproctitle
    HAS_SETPROCTITLE = True
except ImportError:
    HAS_SETPROCTITLE = False
```

```python
def appendproctitle(name):
    '''
    Append "name" to the current process title
    '''
    if HAS_SETPROCTITLE:
        setproctitle.setproctitle(setproctitle.getproctitle() + ' ' + name)
```

This results in wrong main process name. (python instead of salt-minion)

``` 
salt-minion.service - The Salt Minion
   Loaded: loaded (/lib/systemd/system/salt-minion.service; enabled; vendor preset: enabled)
   Active: active (running) since Wed 2017-08-23 22:29:25 UTC; 1 weeks 1 days ago
 Main PID: 916 (/usr/bin/python)
   CGroup: /system.slice/salt-minion.service
           ├─ 916 /usr/bin/python /usr/bin/salt-minion
           ├─1187 /usr/bin/python /usr/bin/salt-minion KeepAlive MinionProcessManage
           └─1222 /usr/bin/python /usr/bin/salt-minion KeepAlive MultiprocessingLoggingQueu
```
 
### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)

### Steps to Reproduce Issue
Install Ubuntu 16.04
install salt-minion
install python-setproctitle
execute 'service salt-minion status'

### Versions Report
Salt Version:
           Salt: 2016.3.7

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.4.2
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.21.1
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.12 (default, Nov 19 2016, 06:48:10)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: Ubuntu 16.04 xenial
        machine: x86_64
        release: 4.4.0-92-generic
         system: Linux
        version: Ubuntu 16.04 xenial

**Comment by gtmanfred** [comment]
Would you mind submitting a PR to fix this in the 2016.11 branch and we can merge it forward to 2017.7 and develop?

Thanks,
Daniel

**Comment by avlasov-mos-de** [comment]
Apparently I wasn't 100% right about root cause. setproctitile() wont accept NULL character. And looks like setprotitile does something on import that modifies process cmdline.

Healthy node:
```
# cat /proc/901/cmdline
/usr/bin/python/usr/bin/salt-minion

# cat /proc/901/cmdline |tr "\0" "\n"
/usr/bin/python
/usr/bin/salt-minion
```

Problem node(setprotitle installed)
```
# cat /proc/26293/cmdline
/usr/bin/python /usr/bin/salt-minionroot

# cat /proc/26293/cmdline | tr "\0" "\n"
/usr/bin/python /usr/bin/salt-minion

cat /proc/26293/cmdline | tr " " "\n"
/usr/bin/python
/usr/bin/salt-minion
```
